src/basic/parse-util.c

```diff
@@ -732,3 +733,51 @@ int parse_loadavg_fixed_point(const char *s, loadavg_t *ret) {
 
         return store_loadavg_fixed_point(i, f, ret);
 }
+
+int named_pcr(char *pcr, unsigned *n) {
+        typedef enum PcrNumber {
+                CORE_FM,
+                PLAT_CONFIG,
+                EXT_CODE,
+                EXT_FM,
+                BOOT_LD,
+                GPT_PT,
+                POWER_ST,
+                SECURE_BT,
+                LINUX_KN = 9,
+                IMA_ST,
+                SYSTEMD_KN,
+                SYSTEMD_BT,
+                SYSTEMD_STB,
+                SHIM_CT,
+                LUKS_KEY,
+                ALIASES_MAX,
+                PCR_INVALID = -EINVAL,
+} PcrNumber;
+
+        static const char* const pcr_number_table[ALIASES_MAX] = {
+                [CORE_FM]     = "corefirmware",
+                [PLAT_CONFIG] = "platformconfig",
+                [EXT_CODE]    = "extcode",
+                [EXT_FM]      = "extfirmware",
+                [BOOT_LD]     = "bootloader",
+                [GPT_PT]      = "gptpartition",
+                [POWER_ST]    = "powerstate",
+                [SECURE_BT]   = "secureboot",
+                [LINUX_KN]    = "linuxkernel",
+                [IMA_ST]      = "imastate",
+                [SYSTEMD_KN]  = "systemdkernel",
+                [SYSTEMD_BT]  = "systemdboot",
+                [SYSTEMD_STB] = "systemdstub",
+                [SHIM_CT]     = "shimcerts",
+                [LUKS_KEY]    = "lukskey",
+        };
+DEFINE_STRING_TABLE_LOOKUP_FROM_STRING(pcr_number, PcrNumber);
```

Comment:
define these 3 outside of the function

Reply:
Wow...
Thank you sir.
I'm currently away from my work space.

I'll push the fix in 2 hours or less.

Just to be sure sir.

The table and enum should both go outside the function, but can be in thesame module still?

Reply:
yes